### PR TITLE
CFE-1099: update ansible-operator-plugins configuration

### PR DESCRIPTION
**Pull request by chiragkyal**
-  Remove `make test-all` and `make test-static` due to redundancy with other tests. 
- Update `make test-unit` to provide permission.

### CODE REVIEW

File: ci-operator/config/openshift/ansible-operator-plugins/openshift-ansible-operator-plugins-main.yaml [hunk 1]
```diff
@@ -43,12 +43,6 @@ tests:
   commands: make test-unit
   container:
     from: src
-- as: all
-  commands: |
-    # Set $HOME for golangci-lint cache
-    HOME=/tmp make test-all
-  container:
-    from: src
```
[the remark this answers is not in the document]
Comment (chiragkyal) [reply ci-operator/config/openshift/ansible-operator-plugins/openshift-ansible-operator-plugins-main.yaml:51]: Oh, I thought the automation would take care of removing from 4.17 and 4.18. Updated now.

File: ci-operator/config/openshift/ansible-operator-plugins/openshift-ansible-operator-plugins-main.yaml [hunk 2]
```diff
@@ -40,19 +40,9 @@ resources:
       memory: 200Mi
 tests:
 - as: unit
-  commands: make test-unit
-  container:
-    from: src
-- as: all
   commands: |
-    # Set $HOME for golangci-lint cache
-    HOME=/tmp make test-all
```
Comment (TrilokGeer) [inline ci-operator/config/openshift/ansible-operator-plugins/openshift-ansible-operator-plugins-main.yaml:49]: I believe, the test targets should be similar to the ones configured for 4.16 release. Is there any test case that fails from [4.16 release ci configuration](https://github.com/openshift/release/blob/f4d243a3c68011e184267fb8126e164dce65bfc1/ci-operator/config/openshift/ocp-release-operator-sdk/openshift-ocp-release-operator-sdk-release-4.16.yaml#L114)?

File: ci-operator/jobs/openshift/ansible-operator-plugins/openshift-ansible-operator-plugins-main-presubmits.yaml [hunk 3]
```diff
@@ -1,58 +1,5 @@
 presubmits:
   openshift/ansible-operator-plugins:
-  - agent: kubernetes
-    always_run: true
-    branches:
-    - ^main$
-    - ^main-
-    cluster: build03
```
Comment (TrilokGeer) [inline ci-operator/jobs/openshift/ansible-operator-plugins/openshift-ansible-operator-plugins-main-presubmits.yaml:8]: These changes might be required for downstream e2e.

Comment (arkadeepsen) [reply ci-operator/jobs/openshift/ansible-operator-plugins/openshift-ansible-operator-plugins-main-presubmits.yaml:8]: For each CI job, these configurations are auto-generated. You can see these configs for `sanity` test: https://github.com/openshift/release/pull/55073/files#diff-96f402a83eeb8a8f09207cf117d2eeabae531803d55e7fe0b2914d98bdb8b0beR57-R109

Comment (arkadeepsen) [reply ci-operator/jobs/openshift/ansible-operator-plugins/openshift-ansible-operator-plugins-main-presubmits.yaml:8]: When e2e test CI config will be added, these same configs will again be auto-generated for the e2e CI job.

**Pull request merged**